pkg/kubelet/kuberuntime/logs/logs_test.go

```diff
@@ -210,6 +212,144 @@ func TestReadLogs(t *testing.T) {
 	}
 }
 
+type syncBuffer struct {
+	b bytes.Buffer
+	sync.Mutex
+}
+
+func (b *syncBuffer) Read(p []byte) (n int, err error) {
+	b.Lock()
+	defer b.Unlock()
+	return b.b.Read(p)
+}
+func (b *syncBuffer) Write(p []byte) (n int, err error) {
+	b.Lock()
+	defer b.Unlock()
+	return b.b.Write(p)
+}
+func (b *syncBuffer) String() string {
+	b.Lock()
+	defer b.Unlock()
+	return b.b.String()
+}
+
+func TestReadRotatedLog(t *testing.T) {
+	tmpDir, err := os.MkdirTemp("", "TestRotatedLog")
+	if err != nil {
+		t.Fatalf("unable to create temp dir")
+	}
+	defer os.RemoveAll(tmpDir)
+
+	file, err := os.CreateTemp(tmpDir, "logfile")
+	if err != nil {
+		t.Fatalf("unable to create temp file")
+	}
+
+	fileName := file.Name()
+	defer os.Remove(fileName)
+
+	file.WriteString(`{"log":"line1\n","stream":"stdout","time":"2023-02-14T11:18:01.00000000Z"}` + "\n")
+	file.WriteString(`{"log":"line2\n","stream":"stdout","time":"2023-02-14T11:18:02.00000000Z"}` + "\n")
+	file.WriteString(`{"log":"line3\n","stream":"stdout","time":"2023-02-14T11:18:03.00000000Z"}` + "\n")
+	// Rotate marker
+	file.WriteString(`{"log":"__ROTATE_MARKER__\n","stream":"stderr","time":"2023-02-14T11:18:04.00000000Z"}` + "\n")
+
+	podLogOptions := v1.PodLogOptions{
+		Follow: true,
+	}
+
+	containerID := "fake-container-id"
+	fakeRuntimeService := &apitesting.FakeRuntimeService{
+		Containers: map[string]*apitesting.FakeContainer{
+			containerID: {
+				ContainerStatus: runtimeapi.ContainerStatus{
+					State: runtimeapi.ContainerState_CONTAINER_RUNNING,
+				},
+			},
+		},
+	}
+
+	opts := NewLogOptions(&podLogOptions, time.Now())
+	stdoutBuf := &syncBuffer{}
+	stderrBuf := &syncBuffer{}
+
+	expected := "line1\nline2\nline3\nline4\nline5\n"
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	evaluated, err := filepath.EvalSymlinks(file.Name())
```

Comment:
Do we really need to evaluate symlinks for the file that we've just created and know that it's not a symlink?